blog/_src/posts/2017-06-06-spring-2017-pl-junior-retrospective.md

```diff
@@ -0,0 +1,162 @@
+for program synthesis.
+
+An additional problem is that it was insufficiently motivated. Either
+skipping the topic or spending more time on it would be preferable,
+since one week was just enough to describe the sequent calculus but
+not enough to apply it.
+
+## Logic Programming
+
+The topic was presented over two weeks. The first session
+presented/demoed Prolog as a language, and we got a sense of what
+logic programming could do. But it was a whirlwind tour, and we were
+left wondering about specific details (how proof search runs, what
+`cut` does).
+
+The second session presented the paper
+[The Semantics of Predicate Logic as a Programming Language][predlogic].
+It was interesting and insightful but left wondering how it relates to
+the implementation of real logic programming languages.
+
+In hindsight this was about as far as we could have gotten in just two
+weeks.
+
+## Bonus Rounds
+
+We also used a few weeks to read and discuss specific papers as a
+group.
+
+The first paper we read was Cardelli's [Typeful Programming][typeful].
+We picked typeful programming because Matthias has mentioned on
+occasion how important he thinks it is.
+
+It was an interesting read; more of an essay than a paper. It really
+stood out as different from the other academic publications that we
+have looked at. It’s a walk through of a language design motivating
+each design decision in practical terms, as in things that actually
+help the programmer.
+
+Cardelli thinks polymorphism (subtyping in addition to parametric) is
+important as well as features for programming in the large such as
+modules and interfaces. Several features are interesting in their
+omission, like type inference and macros.
+
+After reading it it’s not clear why Matthias thinks it’s so important.
+Maybe for historical reasons?
+
+The other paper we read as a group was Alan Kay's
+[The Early History of Smalltalk][smalltalk]. It seems like the
+Smalltalk project investigated a plethora of interesting ideas about
+designing programming languages and environments. This article seems
+to confirm that but does not delve in to many particulars.
+
+## Final Thoughts
+
+Overall this semester of pl junior went well enough that we think it
+makes a reasonable template for future semesters. The topics were
+interesting and relevant, and we mostly picked appropriate material
+for presentations. One downside is that we didn’t quite ‘fill out’ the
+semester with presentations due to scheduling and not wanting to make
+some people present twice. Here’s a lesson: recruit more people to the
+phd program (or get more undergrads) so you don’t have this problem!
+
+Having papers in a theme helped a lot over previous paper-presentation
+iterations of pl junior. It helped each week being able to build on
+what we learned last week, as opposed to having a whirlwind of
+unrelated topics.
+
+Writing this retrospective has proven to be a beneficial exercise.
```

Comment:
Maybe change to "has also proven to be a beneficial exercise. Especially with our sequences of connected topics, looking back has allowed us to put the earlier papers into perspective and better assess both their relevance and presentation." I think it's better if we explain how our retrospective helped.